ax/modelbridge/transforms/unit_x.py

```diff
@@ -109,102 +110,118 @@ def untransform_observation_features(
-                    )
-                    mean = np.asarray(dist.distribution_parameters.get("mean", 0.0))
-                    if not np.all(mean == 0):
-                        if mean.shape != (n_dist_params,):
-                            raise UserInputError(
-                                "Expected `mean` to be an array of shape "
-                                f"{(n_dist_params,)}, but received {mean}."
-                            )
-                        # Mean would simply be AM as long as it is not environmental.
-                        # If environmental, we need to first subtract the lower bounds
-                        # then add the target lb.
-                        if search_space.is_environmental:
-                            lbs = np.array([self.bounds[p][0] for p in dist.parameters])
-                            new_mean = scale_vec * (mean - lbs) + self.target_lb
-                            dist.distribution_parameters["mean"] = new_mean
-                        else:
-                            dist.distribution_parameters["mean"] = scale_vec * mean
-                    continue
-                raise UnsupportedError(
-                    f"{self.__class__.__name__} transform of multivariate "
-                    "distributions, other than `multivariate_normal`, is not "
-                    "supported. Consider manually normalizing the parameter "
-                    "and the corresponding distribution."
+            if len(distribution.parameters) != 1:
+                self._transform_multivariate_distribution(
+                    distribution=distribution, is_environmental=is_environmental
                 )
-            bounds = self.bounds[dist.parameters[0]]
-            p_range = bounds[1] - bounds[0]
-            if p_range == self.target_range and (
-                not search_space.is_environmental or bounds[0] == self.target_lb
-            ):
-                # NOTE: This helps avoid raising the error below if using a discrete
-                # distribution in cases where we do not need to transform.
-                continue
-            loc = dist.distribution_parameters.get("loc", 0.0)
-            if search_space.is_environmental:
-                loc = self._normalize_value(loc, bounds)
             else:
-                loc = loc / p_range * self.target_range
-            dist.distribution_parameters["loc"] = loc
-            dist.distribution_parameters["scale"] = (
-                dist.distribution_parameters.get("scale", 1.0)
-                / p_range
-                * self.target_range
+                self._transform_univariate_distribution(
+                    distribution=distribution, is_environmental=is_environmental
+                )
+
+    def _transform_univariate_distribution(
+        self, distribution: ParameterDistribution, is_environmental: bool
+    ) -> None:
+        r"""Transform a univariate distribution in-place."""
```

Comment:
Is there a reason to do this in-place rather than return a new object?

Reply:
The general pattern in Ax transforms is to do the transforms in-place. So, I just went with that. I think it also leads to a simpler implementation.